operators/src/pro/cache/tile_cache.rs

```diff
@@ -435,218 +809,59 @@ impl TileCache {
     /// Query the cache and on hit create a stream of tiles
     pub async fn query_cache<T>(
         &self,
-        key: CanonicOperatorName,
+        key: &CanonicOperatorName,
         query: &RasterQueryRectangle,
     ) -> Option<CacheTileStream<T>>
     where
         T: Pixel + Cachable,
     {
         let mut backend = self.backend.write().await;
-
-        let (result, expired_entries) = {
-            // lookup the corresponding cache
-            let cache = backend.operator_caches.get_mut(&key)?;
-
-            let FindMatchResult {
-                match_found,
-                expired_entries,
-            } = cache.find_match(query);
-
-            let result = match_found.map(|entry_id| {
-                // if match was found, get the tile stream
-                (
-                    entry_id,
-                    cache
-                        .entries
-                        .get(&entry_id)
-                        .expect("entry should exist because it was returned from `find_match`")
-                        .tile_stream(query),
-                )
-            });
-
-            (result, expired_entries)
-        };
-
-        // remove expired entries from lru index
-        for entry in expired_entries {
-            backend.lru.pop_entry(&entry);
-        }
-
-        // unpack match if there is one
-        let (entry_id, typed_stream) = result?;
-
-        // set as most recently used
-        backend.lru.promote(&entry_id);
-
-        T::stream(typed_stream)
+        backend.query_cache::<T>(key, query)
```

Comment:
if you move all the logic to the backend, do we really need the `TileCache` type anymore?

Reply:
i don't know. Lets keep it until the next refactoring round. It makes things a bit more easy to read. The compiler will take care of all the overhead...